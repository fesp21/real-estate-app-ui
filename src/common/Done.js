import React, { Component, PropTypes } from 'react';
import { View, StyleSheet, StatusBar, Text, TouchableOpacity } from 'react-native';
import Colors from './../common/Colors';
import Ionicons from 'react-native-vector-icons/Ionicons';


export default class Done extends Component {

  handlePress() {
    this.props.emitter.emit('reset');
  }

  render() {
    const { visible } = this.props;
    if (!visible) return null;

    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={() => this.handlePress()}>
          <Text style={styles.title}>Done</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 15,
    paddingRight: 10,
  },
  title: {
    color: Colors.tomato,
    fontSize: 15,
  },
  icon: {
    width: 30,
    height: 30,
    alignSelf: 'center',
    color: Colors.tomato,
  },
  badgeContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },

  badge: {
    backgroundColor: '#fff',
    height: 28,
    width: 28,
    borderRadius: 14,
    margin: 8,
    justifyContent: 'center',
  },

  badgeText: {
    backgroundColor: 'transparent',
    marginTop: -1,
    color: '#0084FF',
    fontSize: 13,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
