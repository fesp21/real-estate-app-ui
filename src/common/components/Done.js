import React, { Component, PropTypes } from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
import colors from './../colors';

export default class Done extends Component {

  static propTypes = {
    emitter: PropTypes.object.isRequired,
    title:PropTypes.string.isRequired
  };

  handlePress() {
    this.props.emitter.emit('reset');
  }

  render() {
    const { visible,title } = this.props;
    if (!visible) return null;

    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={() => this.handlePress()}>
          <Text style={styles.title}>{title}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 15,
    paddingRight: 10,
    backgroundColor:'transparent',
  },
  title: {
    color: colors.tomato,
    fontSize: 15,
  },
  icon: {
    width: 30,
    height: 30,
    alignSelf: 'center',
    color: colors.tomato,
  },
  badgeContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  badge: {
    backgroundColor: '#fff',
    height: 28,
    width: 28,
    borderRadius: 14,
    margin: 8,
    justifyContent: 'center',
  },
  badgeText: {
    backgroundColor: 'transparent',
    marginTop: -1,
    color: '#0084FF',
    fontSize: 13,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
