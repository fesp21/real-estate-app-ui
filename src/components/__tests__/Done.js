import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import Done from "./../Done";

const emitter = {
  emit: jest.fn().mockReturnThis()
};

test("renders null", () => {
  const tree = renderer
    .create(<Done emitter={emitter} visible={true} title="dummy title" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("shows Hello World", () => {
  const component = renderer.create(
    <Done emitter={emitter} visible={true} title="dummy title" />
  );
  let componentJSON = component.toJSON();
  expect(componentJSON.children[0].children[0].children[0]).toBe("dummy title");
});
