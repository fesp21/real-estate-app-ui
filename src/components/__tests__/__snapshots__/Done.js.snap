exports[`test renders null 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingLeft": 15,
      "paddingRight": 10,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#d60c3f",
          "fontSize": 15,
        }
      }>
      dummy title
    </Text>
  </View>
</View>
`;
